Engine/Code/Engine/Graphics/Renderer.cpp

```diff
@@ -249,12 +251,20 @@ void Renderer::Drawtext(const Vec2& position, const Vec4& color, const std::stri
 
 		uvPos = font->GetGlyphUV(asciiText[i]);
 
-		mb->Push(VertexMaster(Vec3(quadPos.m_Mins.m_X, quadPos.m_Mins.m_Y, 0.0f), Vec4(color.m_X, color.m_Y, color.m_Z, color.m_W), Vec2(uvPos.m_Mins.m_X, uvPos.m_Maxs.m_Y)));
-		mb->Push(VertexMaster(Vec3(quadPos.m_Maxs.m_X, quadPos.m_Mins.m_Y, 0.0f), Vec4(color.m_X, color.m_Y, color.m_Z, color.m_W), Vec2(uvPos.m_Maxs.m_X, uvPos.m_Maxs.m_Y)));
-		mb->Push(VertexMaster(Vec3(quadPos.m_Maxs.m_X, quadPos.m_Maxs.m_Y, 0.0f), Vec4(color.m_X, color.m_Y, color.m_Z, color.m_W), Vec2(uvPos.m_Maxs.m_X, uvPos.m_Mins.m_Y)));
-		mb->Push(VertexMaster(Vec3(quadPos.m_Mins.m_X, quadPos.m_Maxs.m_Y, 0.0f), Vec4(color.m_X, color.m_Y, color.m_Z, color.m_W), Vec2(uvPos.m_Mins.m_X, uvPos.m_Mins.m_Y)));
-		
-		Mesh* mesh = mb->CreateMesh<VertexMaster::PCU>();
+		mb->Color3f(Vec4(color.m_X, color.m_Y, color.m_Z, color.m_W));
+		mb->TexCoord2f(Vec2(uvPos.m_Mins.m_X, uvPos.m_Maxs.m_Y));
+	    mb->Position3f(Vec3(quadPos.m_Mins.m_X, quadPos.m_Mins.m_Y, 0.0f));
+	    
+	    mb->TexCoord2f(Vec2(uvPos.m_Maxs.m_X, uvPos.m_Maxs.m_Y));
+	    mb->Position3f(Vec3(quadPos.m_Maxs.m_X, quadPos.m_Mins.m_Y, 0.0f));
+	    
```

Comment:
You don't need to recreate Vec2s just pass uvpos.m_Maxs same for positions since you initialize Zs in 2D to 0. SAme thing for the entire code block pretty much since you are doing copy constructs anyway

Reply:
If it were up to me I would just have a Color3f that takes individual components and do the same for the others as well. @neortls007idev : This assumption only works if the constructor is not marked "explicit" which in this case it would be since I recall asking him to make constructors explicit.

Reply:
But in this case its up to @anishvabardhan to decide. Not my engine.

Reply:
The constructors are explicit